portal-web/test/functional/com/liferay/portalweb/macros/json/layout/JSONLayoutUtil.macro

```diff
@@ -10,13 +10,10 @@ definition {
 
 		Variables.assertDefined(parameterList = "${typeSettings}");
 
+		var widget = JSONLayoutUtil._generateWidget(widgetName = "${widgetName}");
+
 		if (isSet(remoteAppEntryId)) {
-			var widget = JSONLayoutUtil._generateWidget(
-				remoteAppEntryId = "${remoteAppEntryId}",
-				widgetName = "${widgetName}");
-		}
-		else {
-			var widget = JSONLayoutUtil._generateWidget(widgetName = "${widgetName}");
+			var widget = StringUtil.replace("${widget}", "remoteAppEntryId", "${remoteAppEntryId}");
```

Comment:
I think a better method to use is `StringUtil.add`. But looks fine to forward/merge for now and can follow up with refactor later if you get the chance.